src/models/Freeform.js

```diff
@@ -0,0 +1,43 @@
+import Victor from 'victor'
+import Shape, { shapeOptions } from './Shape'
+
+const options = {
+  ...shapeOptions,
+  ...{
+    freeformPoints: {
+      title: 'Points',
+      type: 'input'
+    }
+  }
+}
+
+export default class Freeform extends Shape {
+  constructor() {
+    super('Freeform')
+  }
+
+  getInitialState() {
+    return {
+      ...super.getInitialState(),
+      ...{
+        type: 'freeform',
+        freeformPoints: '-1,-1;-1,1;1,1',
```

Comment:
What if choosing the freeform shape opened a modal (like the import) with a little window you could click into and make a new shape? It wouldn't be editable (unless we could add a button to reopen the modal). And you wouldn't be able to edit it, while looking at it's output transformed, warped, looped. But this seems like an easier way to add this feature.

Regardless, this should probably be in a separate PR.

Reply:
I like that. As it stands now, it's useful for debugging boundary conditions (it was invaluable when coding the inverted masks), but it's commented out of general use.